test/lib/dashycode.js

```diff
@@ -56,15 +56,15 @@ describe('Dashycode', function () {
 	};
 
 	it('should encode all codepoints uniquely', function () {
-		return [...ascii, ...iso88591, ...utf16].reduce((p, codepoint) => (
-			p.then(v => encode(codepoint))
-		), Promise.resolve());
+		for (const codepoint of [...ascii, ...iso88591, ...utf16]) {
+			encode(codepoint);
+		}
 	});
 
 	it('should decode all codepoints accurately', function () {
-		return [...encoded.keys()].reduce((p, dashycode) => (
-			p.then(v => decode(dashycode))
-		), Promise.resolve());
+		for (const dashycode of encoded.keys()) {
+			decode(dashycode);
+		}
```

Comment:
@Kaiepi I don't like `reduce` and this is exactly why. This is 800x more readable as a for-loop.

Reply:
Would use of a Promise would mean that the assertion would have run after the test had finished, so a test failure wouldn't even show up? Also, can you write something like:
```suggestion
		return [...encoded.keys()].reduce((p, dashycode) => (
			p.then(v => decode(dashycode))
		), Promise.resolve());
		for (const [dashycode] of encoded) {
			decode(dashycode);
		}
```

Reply:
@urkerab Promise here is misleading, `decode`/`encode` isn't even an async function.

Reply:
I didn't mean to suggest it was; I was just concerned as to whether the Promise was making the code execute at the wrong time, thus making the test useless.

Reply:
It's been a long time since I wrote this... basically:

> Promise here is misleading.

The point of the promises here is unrelated to `encode`/`decode`. These tests are exhaustive. IIRC, these were prone to blocking other tests when I first wrote it. With a rough benchmark (25 runs of `npx mocha -g '(en|de|trans)cod|att'` in a row), if the promises make any difference, they're within the margin of error, so I'm fine with them going.

> I was just concerned as to whether the Promise was making the code execute at the wrong time, thus making the test useless.

Mocha supports [promise-based](https://mochajs.org/#working-with-promises) asynchronous tests.

> I don't like reduce and this is exactly why. This is 800x more readable as a for-loop.

This for loop isn't equivalent to the reduce... as a for loop, that would look something like:

```javascript
let p = Promise.resolve();
for (const dashycode of encoded.keys()) {
	p = p.then(v => decode(dashycode));
}
return p;
```

To me, this doesn't read better than it did before. JavaScript isn't something I work with very often nowadays though, so I'll take your word for it.

Reply:
@Kaiepi That's not how JavaScript promises work, making a promise doesn't spawn a new thread, this isn't Raku :).

Reply:
@Kaiepi What xfix said. Your implementation does not actually yield. It's exactly the same as my code, only slower.

If you really wanted to yield, you could do it like this:

```ts
const yield = () => new Promise<void>(resolve => {
	setTimeout(() => resolve(), 0);
});

for (const dashycode of encoded.keys()) {
	await yield();
	decode(dashycode);
}
```

Yielding is also pointless in tests; there's nothing to yield to. It's not like we're receiving messages from users or anything.